src/client_metrics.rs

```diff
@@ -167,6 +169,25 @@ pub struct MetricsMetadata {
     pub platform_version: Option<String>,
 }
 
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Builder)]
+#[cfg_attr(feature = "openapi", derive(utoipa::ToSchema))]
+#[serde(rename_all = "camelCase")]
+pub struct MetricSample {
+    pub value: f64,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub labels: Option<HashMap<String, String>>,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Builder)]
+#[cfg_attr(feature = "openapi", derive(utoipa::ToSchema))]
+#[serde(rename_all = "camelCase")]
+pub struct ImpactMetric {
+    pub name: String,
+    pub help: String,
+    pub r#type: String,
```

Comment:
reserved keyword

Reply:
currently type is 'counter' | 'gauge'. Not sure if there's value in Edge being aware of very specific literals

Reply:
> Not sure if there's value in Edge being aware of very specific literals

Nope, not a thing in Rust unfortunately, what you have here is as good as it gets